core/src/main/java/org/apache/iceberg/actions/SortStrategy.java

```diff
@@ -48,7 +49,7 @@ public abstract class SortStrategy extends BinPackStrategy {
    * @return this for method chaining
    */
   public SortStrategy sortOrder(SortOrder order) {
-    this.sortOrder = order;
+    this.sortOrder = SortOrderUtil.buildSortOrder(table(), order);
```

Comment:
This builds a sort order using both the table partition spec and the user supplied sort order.

Reply:
This looks correct to me.

Reply:
Yes, this does happen. We can technically suppress the second occurrence of `X`, but it seems minor to me since it is just going to make the sort use a bit more data. I think this is something that we can clean up in `buildSortOrder` in the future, after this PR.

Reply:
Sounds good!